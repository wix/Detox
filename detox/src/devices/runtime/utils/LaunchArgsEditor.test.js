// @ts-nocheck
describe('Device launch-args editor', () => {
  /** @type {typeof import('./LaunchArgsEditor')} */
  let LaunchArgsEditor;
  /** @type {LaunchArgsEditor} */
  let launchArgsEditor;

  beforeEach(() => {
    LaunchArgsEditor = require('./LaunchArgsEditor');
    launchArgsEditor = new LaunchArgsEditor();
  });

  it('should have empty value at first', () => {
    expect(launchArgsEditor.get()).toEqual({});
  });

  it('should not throw if modify() is called with undefined', () => {
    expect(launchArgsEditor.modify().get()).toEqual({});
  });

  it('should merge values', () => {
    expect(launchArgsEditor
      .modify({ a: 1 })
      .modify({ b: 2 })
      .get()).toEqual({ a: 1, b: 2 });
  });

  it('should merge both local and shared values', () => {
    launchArgsEditor
      .modify({ a: 1 })
      .shared.modify({ b: 2 });

    expect(launchArgsEditor.get()).toEqual({ a: 1, b: 2 });
  });

  it('should not return transient values if permanent values were requested', () =>
    expect(launchArgsEditor
      .modify({ a: 1 })
      .shared.modify({ b: 2 })
      .get()).toEqual({ b: 2 }));

  it('should reset only transient values by default', () => {
    launchArgsEditor.shared.modify({ b: 2 });

    expect(launchArgsEditor
      .modify({ a: 1 })
      .reset()
      .get()).toEqual({ b: 2 });
  });

  it('should reset both permanent and transient values if requested', () => {
    launchArgsEditor.modify({ a: 1 }).shared.modify({ b: 2 });
    launchArgsEditor.reset().shared.reset();
    expect(launchArgsEditor.get()).toEqual({});
  });

  it('should return every time a copy of values', () => {
    expect(launchArgsEditor.get()).not.toBe(launchArgsEditor.get());
    expect(launchArgsEditor.shared.get()).not.toBe(launchArgsEditor.shared.get());
  });

  it('should return every time a deep copy of values', () => {
    launchArgsEditor.modify({ a: { b: 'c' } });
    launchArgsEditor.get().a.b = 'd';
    expect(launchArgsEditor.get()).toEqual({ a: { b: 'c' } });

    launchArgsEditor.shared.modify({ a: { b: 'c' } });
    launchArgsEditor.shared.get().a.b = 'd';
    expect(launchArgsEditor.shared.get()).toEqual({ a: { b: 'c' } });
  });
});
