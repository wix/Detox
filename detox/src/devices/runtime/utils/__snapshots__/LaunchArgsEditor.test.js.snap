// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Device launch-args editor deprecation warnings should log warnings on .get(options) call 1`] = `
"Detected an attempt to use device.launchArgs.get(...) with a deprecated 'options' parameter.
Please use device.appLaunchArgs.get(...) instead.
For more details, see the reference at: https://wix.github.io/Detox/docs/api/device-object-api#deviceapplaunchargs
"
`;

exports[`Device launch-args editor deprecation warnings should log warnings on .get(options) call 2`] = `
"Detected an attempt to use device.launchArgs.get(...) with a deprecated 'options' parameter.
Please use device.appLaunchArgs.shared.get(...) instead.
For more details, see the reference at: https://wix.github.io/Detox/docs/api/device-object-api#deviceapplaunchargs
"
`;

exports[`Device launch-args editor deprecation warnings should log warnings on .modify(..., options) call 1`] = `
"Detected an attempt to use device.launchArgs.modify(...) with a deprecated 'options' parameter.
Please use device.appLaunchArgs.modify(...) instead.
For more details, see the reference at: https://wix.github.io/Detox/docs/api/device-object-api#deviceapplaunchargs
"
`;

exports[`Device launch-args editor deprecation warnings should log warnings on .modify(..., options) call 2`] = `
"Detected an attempt to use device.launchArgs.modify(...) with a deprecated 'options' parameter.
Please use device.appLaunchArgs.shared.modify(...) instead.
For more details, see the reference at: https://wix.github.io/Detox/docs/api/device-object-api#deviceapplaunchargs
"
`;

exports[`Device launch-args editor deprecation warnings should log warnings on .reset(options) call 1`] = `
"Detected an attempt to use device.launchArgs.reset(...) with a deprecated 'options' parameter.
Please use device.appLaunchArgs.reset(...) instead.
For more details, see the reference at: https://wix.github.io/Detox/docs/api/device-object-api#deviceapplaunchargs
"
`;

exports[`Device launch-args editor deprecation warnings should log warnings on .reset(options) call 2`] = `
"Detected an attempt to use device.launchArgs.reset(...) with a deprecated 'options' parameter.
Please use device.appLaunchArgs.shared.reset(...) instead.
For more details, see the reference at: https://wix.github.io/Detox/docs/api/device-object-api#deviceapplaunchargs
"
`;
